examples/dnn_semantic_segmentation_ex.h

```diff
@@ -145,55 +145,129 @@ template <int N, typename SUBNET> using ares_up   = dlib::relu<residual_up<block
+
+template <typename SUBNET> using utag0_ = dlib::add_tag_layer<2110+0,SUBNET>;
+template <typename SUBNET> using utag1_ = dlib::add_tag_layer<2110+1,SUBNET>;
+template <typename SUBNET> using utag2_ = dlib::add_tag_layer<2110+2,SUBNET>;
+template <typename SUBNET> using utag3_ = dlib::add_tag_layer<2110+3,SUBNET>;
+template <typename SUBNET> using utag4_ = dlib::add_tag_layer<2110+4,SUBNET>;
+
+template <typename SUBNET> using concat_utag0 = resize_and_concat<utag0,utag0_,SUBNET>;
+template <typename SUBNET> using concat_utag1 = resize_and_concat<utag1,utag1_,SUBNET>;
+template <typename SUBNET> using concat_utag2 = resize_and_concat<utag2,utag2_,SUBNET>;
+template <typename SUBNET> using concat_utag3 = resize_and_concat<utag3,utag3_,SUBNET>;
+template <typename SUBNET> using concat_utag4 = resize_and_concat<utag4,utag4_,SUBNET>;
 
-template <typename SUBNET> using level1t = dlib::repeat<2,res512,res_up<512,SUBNET>>;
-template <typename SUBNET> using level2t = dlib::repeat<2,res256,res_up<256,SUBNET>>;
-template <typename SUBNET> using level3t = dlib::repeat<2,res128,res_up<128,SUBNET>>;
-template <typename SUBNET> using level4t = dlib::repeat<2,res64,res_up<64,SUBNET>>;
+// ----------------------------------------------------------------------------------------
 
-template <typename SUBNET> using alevel1t = dlib::repeat<2,ares512,ares_up<512,SUBNET>>;
-template <typename SUBNET> using alevel2t = dlib::repeat<2,ares256,ares_up<256,SUBNET>>;
-template <typename SUBNET> using alevel3t = dlib::repeat<2,ares128,ares_up<128,SUBNET>>;
-template <typename SUBNET> using alevel4t = dlib::repeat<2,ares64,ares_up<64,SUBNET>>;
+// Microsoft Visual C++ compilers choke when trying to build networks that are too deep.
+// So let's limit the depth a little. On the PASCAL VOC 2012 data, the accuracy hit is
+// not even very significant.
 
-// ----------------------------------------------------------------------------------------
+#ifdef _MSC_VER
+#define LIMIT_LAYER_COUNT_BECAUSE_OF_COMPILER_ISSUES 1
+static const char* semantic_segmentation_net_filename = "semantic_segmentation_voc2012net_v2_limited_layer_count.dnn";
+#else
+#define LIMIT_LAYER_COUNT_BECAUSE_OF_COMPILER_ISSUES 0
+static const char* semantic_segmentation_net_filename = "semantic_segmentation_voc2012net_v2.dnn";
+#endif
 
 // training network type
-using net_type = dlib::loss_multiclass_log_per_pixel<
-                            dlib::cont<class_count,7,7,2,2,
-                            level4t<level3t<level2t<level1t<
-                            level1<level2<level3<level4<
-                            dlib::max_pool<3,3,2,2,dlib::relu<dlib::bn_con<dlib::con<64,7,7,2,2,
-                            dlib::input<dlib::matrix<dlib::rgb_pixel>>
-                            >>>>>>>>>>>>>>;
+using bnet_type = dlib::loss_multiclass_log_per_pixel<
+                              dlib::cont<class_count,1,1,1,1,
+#if LIMIT_LAYER_COUNT_BECAUSE_OF_COMPILER_ISSUES == 0
```

Comment:
It's a little easier to read if written `#ifdef LIMIT_LAYER_COUNT_BECAUSE_OF_COMPILER_ISSUES`

Reply:
Or actually `#ifndef LIMIT_LAYER_COUNT_BECAUSE_OF_COMPILER_ISSUES` -- but as I wrote, I'd like to get rid of that anyway.